src/hooks/useClickOutside.js

```diff
@@ -0,0 +1,15 @@
+import { useEffect, useRef } from "react";
+
+export const useClickOutside = (handler) => {
+  const domNode = useRef();
+  useEffect(() => {
+    const clickHandler = (e) => {
+      if (domNode.current && !domNode.current.contains(e.target)) {
+        handler();
+      }
+    };
+    document.addEventListener("mousedown", clickHandler);
```

Comment:
In react, we normally not mix JS syntax. If there is any alternative, you should use that.

Reply:
This is a different usecase where I was required to access dom directly. I have looked for solutions and this was used in multiple places. Thank you for the review!